cmd/axelard/cmd/vald/tss/tss.go

```diff
@@ -261,11 +277,18 @@ func (mgr *Mgr) ProcessHeartBeatEvent(e tmEvents.Event) error {
 		case exported.Threshold:
 			request = &tofnd.KeyPresenceRequest{
 				KeyUid: string(keyInfo.KeyID),
+				PubKey: []byte{},
 			}
 			response, err = mgr.client.KeyPresence(grpcCtx, request)
 		case exported.Multisig:
+			pubKey := []byte{}
+			if pubKeys, found := mgr.getPubKeys(keyInfo.KeyID); found {
+				pubKey = pubKeys[0]
+			}
+
 			request = &tofnd.KeyPresenceRequest{
```

Comment:
Why even bother checking with tofnd when vald doesn't even think that it has anything to do with that particular key ID.

Reply:
I was planning to go with a less stateful approach before, but now that I track pub keys from key rotations in vald, this makes sense. In fact, this will actually make heartbeats more useful for multisig. I remembered that all validators signal key presence for all key ids, even if they aren't in the snapshot for that key id (this was a downside of tofnd being stateless).